packages/kerosene-test/src/react/createStubComponent.tsx

```diff
@@ -0,0 +1,91 @@
+import { identity } from "lodash";
+import * as React from "react";
+
+export default function createStubComponent<
+  Props extends {} = { [prop: string]: unknown },
+  RenderProps extends any[] = any[]
+>(displayName: string, functional?: false): React.ComponentClass<Props>;
+
+export default function createStubComponent<
+  Props extends {} = { [prop: string]: unknown },
+  RenderProps extends any[] = any[]
+>(displayName: string, functional: true): React.FunctionComponent<Props>;
+
+export default function createStubComponent<
+  Props extends {} = { [prop: string]: unknown },
+  RenderProps extends any[] = any[]
+>(
+  displayName: string,
+  functional: false | undefined,
+  getRenderProps: () => RenderProps,
+  transformRenderPropResult?: (
+    result: (props: any) => React.ReactNode,
+  ) => React.ReactNode,
+): React.ComponentClass<
+  Props & { children: (...args: RenderProps) => React.ReactNode }
+>;
+
+export default function createStubComponent<
+  Props extends {} = { [prop: string]: unknown },
+  RenderProps extends any[] = any[]
+>(
+  displayName: string,
+  functional: true,
+  getRenderProps: () => RenderProps,
+  transformRenderPropResult?: (
+    result: (props: any) => React.ReactNode,
+  ) => React.ReactNode,
+): React.FunctionComponent<
+  Props & { children: (...args: RenderProps) => React.ReactNode }
+>;
+
+/**
+ * Creates a StubComponent for unit testing
+ * @param displayName Component displayName which can be used to `EnzymeWrapper#find` the component
```

Comment:
can these JSDoc comments be "upgraded" to real TS types?

Reply:
What do you mean by that?

Reply:
can you put this type information as actual typescript instead of as jsdoc annotations